### PR TITLE
29-cart page

### PR DESCRIPTION
**Pull request by akashaf**

### CODE REVIEW

File: app/containers/CheckoutPage/saga.js [hunk 1]
```diff
@@ -1,13 +1,54 @@
...
+    const body = JSON.stringify({
+        param: action.id,
+        qty: action.qty,
+    });
+    const response = yield call(apiRequest, `/cart/${action.id}`, 'put', body);
+    if (response && response.ok) {
+        yield put(updateQtySuccess(response.data));
+        notifySuccess(response.data.messages[0].text);
+    } else {
+        // yield put(listingRequestFailed(response.data));
```
Comment (tienping) [inline app/containers/CheckoutPage/saga.js]: i think we will need this failed handling

File: app/containers/CheckoutPage/reducer.js [hunk 2]
```diff
@@ -5,16 +5,51 @@
...
+                .set('data', action.r);
+        case QTY_UPDATE:
+            return state
+                .set('loading', true)
+                .set('error', false);
+        case QTY_UPDATE_SUCCESS:
+            return state
+                .set('loading', false)
+                .set('error', false)
+                .set('data', dataChecking(action, 'r', 'cart'));
```
Comment (tienping) [inline app/containers/CheckoutPage/reducer.js]: can u give proper name for r?
e.g. response

Comment (tienping) [reply app/containers/CheckoutPage/reducer.js]: or, payload

**Review by tienping — CHANGES_REQUESTED** [review]

File: app/components/CartPage/index.js [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+*/
+
+import React from 'react';
+import { dataChecking } from 'globalUtils';
+
+import './style.scss';
+
+class CartPage extends React.PureComponent { // eslint-disable-line react/prefer-stateless-function
+
+    is_qty_adjustable = (item) => {
```
Comment (tienping) [inline app/components/CartPage/index.js]: function use camel case

Comment (tienping) [reply app/components/CartPage/index.js]: isQtyAdjustable suppose to return boolean, if this is suppose to return element, use renderQuantityEditor instead

File: app/components/CartPage/index.js [hunk 4]
```diff
@@ -0,0 +1,129 @@
...
+        if (item.attribute.is_qty_adjustable) {
+            return (
+                <div className="text-xs-center" style={{ width: '100px' }}>
+                    <span
+                        className="px-quater"
+                        style={{ cursor: 'pointer' }}
+                        onClick={() => this.props.addQty('remove', item.qty, item.id)}
```
Comment (tienping) [inline app/components/CartPage/index.js]: use this.addQty?

Comment (tienping) [reply app/components/CartPage/index.js]: btw... how can addQty support remove, rename the function to changeQuantity

File: app/containers/CheckoutPage/index.js [hunk 5]
```diff
@@ -8,27 +8,62 @@ import React from 'react';
...
+    componentDidMount() {
+        this.props.dispatch(getCheckout());
+    }
+
+    addQty = (type, qty, id) => {
+        if (qty <= 1 && type === 'remove') {
+            return;
+        }
+        let quantity = qty;
+        quantity = type === 'add' ? quantity += 1 : quantity -= 1;
```
Comment (tienping) [inline app/containers/CheckoutPage/index.js]: quantity += type === 'add' ? 1 : -1;

File: app/containers/CheckoutPage/reducer.js [hunk 6]
```diff
@@ -5,16 +5,51 @@
...
+    loading: false,
+    error: false,
+    data: null,
+});
 
 function checkoutPageReducer(state = initialState, action) {
     switch (action.type) {
-        case DEFAULT_ACTION:
-            return state;
+        case GET_CHECKOUT_DATA:
```
Comment (tienping) [inline app/containers/CheckoutPage/reducer.js:33]: u dint handle for failed at all?

**Review by tienping — CHANGES_REQUESTED** [review]

File: app/components/CartPage/index.js (hunk 4 above)
Comment (akashaf) [reply app/components/CartPage/index.js]: this function is use for update qty. will change it..

**Review by tienping — APPROVED** [review]

**Pull request merged**